### ISSUE TITLE
"invalid opcode" error in /usr/lib32/libopenal.so.1.19.0 when entering Big Picture Mode

### ISSUE THREAD

**Issue by volkertb** [CaptSilver/steamos-clear#7]
Hi! First of all: awesome project you're working on. Thanks for making it easier for everyone to play Steam games on Clear Linux. 🙂

Anyway, I ran into an issue after I had installed Steam on Clear Linux using your script. The application kept crashing whenever I tried to start or go into Big Picture mode. When running `dmesg` after such a crash, I would see "invalid opcode" errors w.r.t. the system library libopenal.so.1.19.0:

```
[ 1659.847109] traps: steam[2338] trap invalid opcode ip:efa1da30 sp:ffd46fe0 error:0 in libopenal.so.1.19.0[ef9d9000+6b000]
```

I found two possible workarounds, both of which come down to forcing Steam to use its bundled version of the openal.so.1 library instead:

1. Delete or move the 32-bit version of the system library. 
    ```
   mkdir -p ~/backup/usr/lib32
   sudo mv /usr/lib32/libopenal.so.1.19.0 ~/backup/usr/lib32/
   ```
   The possible side-effect of this would be that 32-bit non-Steam games might cease to work if they require OpenAL. But most of us use this script to run a dedicated Steam machine, so that might not be too much of an issue. From what I have understood, Steam makes its bundled libraries available to the games it launches.

2. Launch Steam with an LD_PRELOAD environment variable that forces it to load its bundled version of the library before anything else.
   ```
   LD_PRELOAD=~/.local/share/Steam/ubuntu12_32/steam-runtime/i386/usr/lib/i386-linux-gnu/libopenal.so.1 steam nobigpicture
   ```
   You'll see errors appear in the console output claiming that preloading has failed, but it will work nevertheless. This approach would however require the Steam executable to be wrapped in a script that passes this environment variable to it on startup, so that this would also work when directly booting into Big Picture mode. Such a wrapper script would impose another maintenance burden, since this problem might be resolved and/or instead might start to occur with other libraries as both bundled libraries and system libraries get updated from time to time.

Does your script depend on the [linux-steam-integration](https://github.com/clearlinux/linux-steam-integration) project for Clear Linux? I did notice the `lsi-exec` and `lsi-settings` being installed in my Clear Linux system after having installed Steam using your script, but I don't see that project referenced anywhere in your project. If your project indeed does depend on it, do you think I should report this issue there instead?

**Comment by volkertb** [comment]
By the way, I did upgrade to the beta version of the Steam client. That might be a factor.

**Comment by volkertb** [comment]
I guess I could also append `:$HOME/.local/share/Steam/ubuntu12_32/steam-runtime/i386/usr/lib/i386-linux-gnu/libopenal.so.1` to the existing `export LD_PRELOAD=` line in `/usr/bin/steamos-session`. I'm going to check if that works when directly booting into Big Picture Mode.

**UPDATE**: Yes, I ran `reboot-to-steamos-mode` after making the change I suggested above and it works! 😃 I can make a PR if you want. But before I do that, perhaps others can try to reproduce this issue on their hardware first?

**Comment by CaptSilver** [comment]
Indeed. I will have to try out this weekend. I got busy at work, sorry for the late reply.
